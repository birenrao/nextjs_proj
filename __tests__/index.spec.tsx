import * as React from 'react';
import { mount, shallow } from 'enzyme';
import IndexPage from '../pages/index';
import { items } from './items';

let tree: any;

beforeEach(() => {
  tree = shallow(<IndexPage items={items} pathname={''} />);
});
describe('Pages', () => {
  describe('Index', () => {
    it('should render without throwing an error', () => {
      const wrap = mount(<IndexPage />);
      expect(wrap.find('h1').text()).toBe('My Movies');
    });

    it('renders inadex page correctly', () => {
      expect(tree.debug()).toMatchSnapshot();
    });
  });

});
})
