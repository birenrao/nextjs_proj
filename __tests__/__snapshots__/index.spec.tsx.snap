// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Index renders inadex page correctly 1`] = `
"<ContextProvider value={{...}}>
  <Layout>
    <div className=\\"errorMSg\\">
      Error: due to some technical issues movies details are not available  
    </div>
    <p>
      <Link href=\\"/\\">
        <a>
          Go home
        </a>
      </Link>
    </p>
  </Layout>
</ContextProvider>"
`;
